### PR TITLE
For #18142: address bar freeze

### PR DESCRIPTION
**Pull request by MarcLeclair**
For #18142

This fixes the jank created when a user has a super long string in their clipboard and tries to click on the URL bar. 

However, I have to point out that by calling `URLStringUtils`, there is one small change in behavior.

- Behavior before the changes:
If a user had the following string in their clipboard `abcdefg mozilla.com` the clipboard would offer to "FIll the link" with 
`mozilla.com`. Another example would be having `mozilla.com https://mozilla.com` would show the option to paste the link `https://mozilla.com`.

- Behavior with this patch:
If a user has the string `abcdefg mozilla.com`, nothing will be shown. However, if the user has  some sort of URL (ie: `c-c.com`, `about-mozilla:mozilla`, etc...) it will offer to paste the link from the clipboard.

The reason for the change is that to fix the problem in `WebURLFinder.kt`, we would need to fix the regex which is used in multiple different place ( one of them being `TabIntentProcessor.kt`). I didn't want to change the behavior of those classes by changing the regex. 
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by mcomella — COMMENTED** [review]
To be explicit, I think we should get a reviewer from fenix for this change as they'll be in a better position to validate the correctness of this change. I'm doing a skim for perf issues.

This general approach seems good to me for perf but we should squash commits like, "Fixed indentation" into the original commit (assuming it's not too much trouble).

File: app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt [hunk 1]
```diff
@@ -38,8 +38,7 @@ class ClipboardHandler(val context: Context) {
     val url: String?
         get() {
             return text?.let {
-                val finder = WebURLFinder(it)
-                finder.bestWebURL()
+                if(URLStringUtils.isURLLike(it)) it else null
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/utils/ClipboardHandler.kt:41]:
```suggestion
                if (URLStringUtils.isURLLike(it)) { it } else { null }
```
should fix lint issue

**Comment by eliserichards** [comment]
I'll check if the use case you mentioned makes a difference, but I think we should be good to go on this once that's confirmed.

**Comment by eliserichards** [comment]
I spoke to Vesta, and we don't want to regress that behavior. 

What is the tradeoff here if we hold off on merging this until a solution can be found to ensure we still have that clipboard behavior?

**Comment by MarcLeclair** [comment]
@eliserichards  the trade off at this moment is the performance / jankiness that occurs when having a long string in the keyboard is can affect users. We can keep the same behavior by duplicating code (i.e: the method that separates the string into multiple part to identify differenty URL isn't the root of the performance problem).  However, anyone with a long string in their clipboard will take longer to get to the search experience. I'd like to add that this doesn't happen on any other browser that I've tried it on (i.e: having a long string in my clipboard doesn't lag)

**Comment by mcomella** [comment]
This is a possible fix but the discussion around it has stalled so we're going to close it. Feel free to reopen and iterate if it's useful to you.

**Pull request closed without merging**